cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -366,42 +411,60 @@ func (m *azureCache) FindForInstance(instance *azureRef, vmType string) (cloudpr
 		return nil, err
 	}
 	inst := azureRef{Name: resourceID}
+
+	if unowned, err := m.isUnownedInstance(inst, ""); unowned || err != nil {
+		return nil, err
+	}
+
+	if invalid, err := m.isInvalidVMType(inst, vmType, vmsPoolSet); invalid || err != nil {
+		return nil, err
+	}
+
+	// Look up caches for the instance.
+	klog.V(6).Infof("FindForInstance: attempting to retrieve instance %v from cache", m.instanceToNodeGroup)
+	if nodeGroup := m.getInstanceFromCache(inst.Name); nodeGroup != nil {
```

Comment:
need to validate what `getInstanceFromCache` returns, since it seems to be the `nodeGroup`???

Reply:
it returns nodegroup but we don't really care what it returns just that there is an instance with a nodegroup mapping in the cache.